modules/transport/Transport.js

```diff
@@ -0,0 +1,265 @@
+import {
+    MESSAGE_TYPE_EVENT,
+    MESSAGE_TYPE_RESPONSE,
+    MESSAGE_TYPE_REQUEST
+} from './constants';
+
+/**
+ * Stores the currnet transport backend that have to be used. Also implements
+ * request/response mechanism.
+ */
+export default class Transport {
```

Comment:
We are caching unprocessed messages as you can see in the implementation. 

The result of the listeners indicates whether the message is processed by this listener or not. If there is a message not processed by any listener we are storing it and passing it to every new listener added until a listener process it. 

This solves some race condition issues where the consumers of the events haven't yet added their listeners but we have received some messages from outside.